### PR TITLE
Use proper Bitbucket API for permission detection

### PR DESCRIPTION
**Pull request by folex**
Today I faced an issue described here https://discourse.drone.io/t/drone-deploy-with-client-error-404-error/1234/8 and decided to fix it.

I use https://developer.atlassian.com/bitbucket/api/2/reference/resource/user/permissions/repositories with filter by repository `full_name` to find out what permissions user have on that repository.

I've added a few basic tests, please tell me if I should write more tests for this PR.

P.S. I'm pretty new to Go, and would be thankful for any style-wise comments.

### CODE REVIEW

File: remote/bitbucket/internal/client.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 	"net/http"
 	"net/url"
 
+	"github.com/pkg/errors"
```
Comment (bradrydzewski) [inline remote/bitbucket/internal/client.go]: change from `"github.com/pkg/errors"` to `"errors"`

File: remote/bitbucket/internal/client.go [hunk 2]
```diff
@@ -152,6 +154,23 @@ func (c *Client) CreateStatus(owner, name, revision string, status *BuildStatus)
...
+	out := new(RepoPermResp)
+	uri := fmt.Sprintf(pathPermissions, c.base, full_name)
+	err := c.do(uri, get, nil, out)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(out.Values) == 0 {
+		err = errors.New(fmt.Sprint("no permissions in repository ", full_name))
```
Comment (bradrydzewski) [inline remote/bitbucket/internal/client.go]: actually, you can just do `fmt.Errorf("no permissions in repository %s", fullName)`

File: remote/bitbucket/internal/client.go [hunk 3]
```diff
@@ -152,6 +154,23 @@ func (c *Client) CreateStatus(owner, name, revision string, status *BuildStatus)
 	return c.do(uri, post, status, nil)
 }
 
+func (c *Client) GetPermission(full_name string) (*RepoPerm, error) {
```
Comment (bradrydzewski) [inline remote/bitbucket/internal/client.go]: use `fullName` instead of `full_name`

File: remote/bitbucket/internal/client.go [hunk 4]
```diff
@@ -34,15 +35,16 @@ const (
...
-	pathTeams  = "%s/2.0/teams/?%s"
-	pathRepo   = "%s/2.0/repositories/%s/%s"
-	pathRepos  = "%s/2.0/repositories/%s?%s"
-	pathHook   = "%s/2.0/repositories/%s/%s/hooks/%s"
-	pathHooks  = "%s/2.0/repositories/%s/%s/hooks?%s"
-	pathSource = "%s/1.0/repositories/%s/%s/src/%s/%s"
-	pathStatus = "%s/2.0/repositories/%s/%s/commit/%s/statuses/build"
+	pathUser        = "%s/2.0/user/"
+	pathEmails      = "%s/2.0/user/emails"
+	pathPermissions = "%s/2.0/user/permissions/repositories?q=repository.full_name=\"%s\""
```
Comment (bradrydzewski) [inline remote/bitbucket/internal/client.go]: use %q instead

```
pathPermissions = "%s/2.0/user/permissions/repositories?q=repository.full_name=%q"
```

**Comment by bradrydzewski** [comment]
thanks for the pull request!  Some minor comments, but otherwise looks great

**Comment by folex** [comment]
Thanks a lot for comments! I've fixed everything you mentioned.

Should I squash commits to a single one?

**Comment by folex** [comment]
Oops. Fixed it now :)

**Comment by bradrydzewski** [comment]
thanks!

**Pull request merged**